Bump up lib versions

nyandika opened the pull request:
## Changes proposed

There have been recent vulnerabilities in the nextjs space. Taking this chance to bump up all package versions

Also added prod branch to github actions. This necessisated a rerun of prettier and linting hence the other numerous file changes

## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots

N/A

## Note to reviewers

N/A

mosesmbadi committed:
added docker

nyandika committed:
Bump up lib versions

nyandika committed:
Update actions to v4

orama254 reviewed (changes requested):
Hello @nyandika,

Thanks for this very critical pull request. Could you check up on the conflicts emerging?

nyandika commented:
Working on the failing tests. Will send an update

nyandika committed:
refactor

nyandika committed:
Merge prod

nyandika committed:
Fix build

nyandika committed:
Fix failing tests

nyandika committed:
Reset event changes

nyandika committed:
Run prettier

nyandika committed:
Update eslint

nyandika committed:
Update github actions

nyandika committed:
Resolve e2e

nyandika committed:
Refactor

nyandika committed:
Fix font

nyandika committed:
Fix overlay for tests

nyandika commented:
@orama254 Could you authorize this PR to deploy to vercel to get the pre-release view
I have fixed all failing tests

festus-sulumeti committed:
docs(readme): overhaul README with full project documentation and setup guide

festus-sulumeti committed:
chore(security): document npm audit vulnerabilities in dependency tree

festus-sulumeti committed:
feat(ui): setup shadcn/ui components with Tailwind support

festus-sulumeti committed:
feat(ui): add Button component via shadcn/ui

festus-sulumeti committed:
feat: updated codebase

festus-sulumeti committed:
feat(navbar): redesign navbar with responsive menu and prominent blue CTA button

festus-sulumeti committed:
chore: deleted the dropdown menu from the navbar

festus-sulumeti committed:
chore: removed the unused image from the importation

orama254 commented:
@greptile

orama254 committed:
Merge branch 'prod' into refactor/lib-updates

nyandika committed:
Fix build

nyandika committed:
Further updates to libs

nyandika committed:
Add next-env.d.ts to ignore

nyandika committed:
Fix e2e tests

nyandika committed:
Fix format

nyandika commented:
@orama254 fixed issues brought about by merge from prod

orama254 commented:
@greptile

orama254 committed:
Update tailwind.config.js

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

orama254 committed:
Update src/components/HeroHeader/HeroHeader.tsx

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

orama254 committed:
chore: resolve formatting issue

orama254 committed:
chore: lockfile fix

orama254 committed:
chore: remove orphan code

orama254 committed:
chore: ci refactor fox

useState<number | null>(() => typeof window !== 'undefined' ? window.innerWidth : null) — initializes the value immediately via a lazy initializer instead of inside the effect

typeof window !== 'undefined' guard prevents SSR errors during Next.js server rendering

orama254 committed:
chore: prettier code formatting

nyandika committed:
Run e2e tests against prod build

orama254 commented:
LGTM,

Thanks @nyandika for the prompt actioning of the pending tasks
